external_services/lti.py

```diff
@@ -19,8 +20,9 @@ class LTIRequest(object):
     def __init__(self, service, user, instance, request, title, context_id=None, link_id=None, add=None, exercise=None):
         self.service = service
         course = instance.course
+        parsed_url = urlparse(settings.BASE_URL)
         # Context and resource parameters.
-        context_id = context_id or (request.get_host() + instance.get_absolute_url())
+        context_id = context_id or (parsed_url.hostname + ":" + str(parsed_url.port) + instance.get_absolute_url())
```

Comment:
I think `parsed_url.netloc` should be ok here? I think we can assume that ports 80 and 443 (defaults for http and https) always share the same site. However, are you ommitting a sub path (e.g. `http://example.com/aplus/`), or was that handled by the get_absolute_url()?

(again, nice source ref: https://github.com/python/cpython/blob/master/Lib/urllib/parse.py#L199)

Reply:
Yeah, netloc seems very ok, I didn't notice such an option. About the subpath, I'm pretty sure it's handled by get_absolute_url() (works ok at least locally), so it should be good unless I'm missing something.

Reply:
Hmm.. Yeah. It (get_absolute_url) appears to handle this correctly as long as the system admin installs the application correctly.